pkg/autoscaler/autoscaler/util.go

```diff
@@ -88,36 +93,97 @@ func limitTargetReplicas(targetReplicas int32, tac *v1alpha1.TidbClusterAutoScal
+	case v1alpha1.TiKVMemberType:
+		requests = tc.Spec.TiKV.Requests
+	}
+	for res, v := range requests {
+		switch res {
+		case corev1.ResourceCPU:
+			resource.CPU = v
+		case corev1.ResourceMemory:
+			resource.Memory = v
+		case corev1.ResourceStorage:
+			resource.Storage = v
+		}
+	}
+	tac.Spec.Resources[typ] = resource
+}
+
+func defaultResourceTypes(tac *v1alpha1.TidbClusterAutoScaler, rule *v1alpha1.AutoRule, component v1alpha1.MemberType) {
+	if len(rule.ResourceTypes) == 0 {
+		for typ, res := range tac.Spec.Resources {
+			if component == v1alpha1.TiKVMemberType {
+				// The resource has no storage for TiKV
+				if res.Storage.Cmp(zeroQuantity) == 0 {
+					continue
+				}
+			}
+			rule.ResourceTypes = append(rule.ResourceTypes, typ)
+		}
+	}
+}
+
+func defaultBasicAutoScaler(tac *v1alpha1.TidbClusterAutoScaler, spec *v1alpha1.BasicAutoScalerSpec, component v1alpha1.MemberType) {
+	if spec.MinReplicas == nil {
+		spec.MinReplicas = pointer.Int32Ptr(1)
+	}
+	if spec.ScaleOutIntervalSeconds == nil {
+		spec.ScaleOutIntervalSeconds = pointer.Int32Ptr(300)
+	}
+	if spec.ScaleInIntervalSeconds == nil {
+		spec.ScaleInIntervalSeconds = pointer.Int32Ptr(500)
+	}
+	// If ExternalEndpoint is not provided, we would set default metrics
+	if spec.External == nil && spec.MetricsTimeDuration == nil {
+		spec.MetricsTimeDuration = pointer.StringPtr("3m")
+	}
+
+	for res, rule := range spec.Rules {
+		if res == corev1.ResourceCPU {
+			if rule.MinThreshold == nil {
+				rule.MinThreshold = pointer.Float64Ptr(0.1)
+			}
+		}
+		defaultResourceTypes(tac, &rule, component)
+		spec.Rules[res] = rule
+	}
+}
+
 // If the minReplicas not set, the default value would be 1
 // If the Metrics not set, the default metric will be set to 80% average CPU utilization.
 // defaultTAC would default the omitted value
-func defaultTAC(tac *v1alpha1.TidbClusterAutoScaler) {
+func defaultTAC(tac *v1alpha1.TidbClusterAutoScaler, tc *v1alpha1.TidbCluster) {
 	if tac.Annotations == nil {
 		tac.Annotations = map[string]string{}
 	}
 
-	def := func(spec *v1alpha1.BasicAutoScalerSpec) {
-		if spec.MinReplicas == nil {
-			spec.MinReplicas = pointer.Int32Ptr(1)
-		}
-		if spec.ScaleOutIntervalSeconds == nil {
-			spec.ScaleOutIntervalSeconds = pointer.Int32Ptr(300)
+	if len(tac.Spec.Resources) == 0 {
+		// Construct default resource
+		if tac.Spec.TiKV != nil {
+			defaultResources(tc, tac, v1alpha1.TiKVMemberType)
 		}
-		if spec.ScaleInIntervalSeconds == nil {
-			spec.ScaleInIntervalSeconds = pointer.Int32Ptr(500)
-		}
-		// If ExternalEndpoint is not provided, we would set default metrics
-		if spec.External == nil && spec.MetricsTimeDuration == nil {
-			spec.MetricsTimeDuration = pointer.StringPtr("3m")
+
+		if tac.Spec.TiDB != nil {
+			defaultResources(tc, tac, v1alpha1.TiDBMemberType)
 		}
 	}
 
 	if tidb := tac.Spec.TiDB; tidb != nil {
-		def(&tidb.BasicAutoScalerSpec)
+		defaultBasicAutoScaler(tac, &tidb.BasicAutoScalerSpec, v1alpha1.TiDBMemberType)
```

Comment:
The `TiDBMemberType ` should be able to identify the `tidb.BasicAutoScalerSpec`.